### ISSUE TITLE
NullReferenceException when using numeric private key property

### ISSUE THREAD

**Issue by klemmchr** [borisdj/EFCore.BulkExtensions#746]
When upgrading from .NET Core 3.1 (EfCore.BulkExtensions `3.6.5`) to .NET 6 (EfCore.BulkExtensions `6.3.2`) I stumbled upon a `NullReferenceException` when using a private key property.

```
System.NullReferenceException
Object reference not set to an instance of an object.
   at EFCore.BulkExtensions.TableInfo.<>c__DisplayClass171_3`1.<LoadData>b__39(T a) in C:\Users\Foo\EFCore.BulkExtensions\EFCore.BulkExtensions\TableInfo.cs:line 240
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at EFCore.BulkExtensions.TableInfo.LoadData[T](DbContext context, Type type, IList`1 entities, Boolean loadOnlyPKColumn) in C:\Users\Foo\EFCore.BulkExtensions\EFCore.BulkExtensions\TableInfo.cs:line 0
   at EFCore.BulkExtensions.TableInfo.CreateInstance[T](DbContext context, Type type, IList`1 entities, OperationType operationType, BulkConfig bulkConfig) in C:\Users\Foo\EFCore.BulkExtensions\EFCore.BulkExtensions\TableInfo.cs:line 100
   at EFCore.BulkExtensions.DbContextBulkTransaction.Execute[T](DbContext context, Type type, IList`1 entities, OperationType operationType, BulkConfig bulkConfig, Action`1 progress) in C:\Users\Foo\EFCore.BulkExtensions\EFCore.BulkExtensions\DbContextBulkTransaction.cs:line 32
   at EFCore.BulkExtensions.DbContextBulkExtensions.BulkDelete[T](DbContext context, IList`1 entities, BulkConfig bulkConfig, Action`1 progress, Type type) in C:\Users\Foo\EFCore.BulkExtensions\EFCore.BulkExtensions\DbContextBulkExtensions.cs:line 113
   at EFCore.BulkExtensions.Tests.EFCoreBulkTestAtypical.PrivatePropertiesDefaultValue() in C:\Users\Foo\EFCore.BulkExtensions\EFCore.BulkExtensions.Tests\EFCoreBulkTestAtypical.cs:line 692
```

I was able to reproduce this bug using the following configuration and test:

```csharp
public class DefaultValueTest
{
    private long Number { get; set; }
    public string Test { get; set; }
}
```

```csharp
modelBuilder.Entity<DefaultValueTest>(c =>
{
    c.HasKey("Number");
    
    c.Property("Number").ValueGeneratedOnAdd();
});
```

```csharp
[Fact]
private void PrivatePropertiesDefaultValue()
{
    ContextUtil.DbServer = DbServer.SQLServer;
    using (var context = new TestContext(ContextUtil.GetOptions()))
    {
        context.BulkDelete(context.DefaultValueTests.ToList());
    }

    using (var context = new TestContext(ContextUtil.GetOptions()))
    {
        var entities = new List<DefaultValueTest> {
            new()
            {
                Test = "test"
            }
        };
        context.BulkInsertOrUpdate(entities);
    }

    using (var context = new TestContext(ContextUtil.GetOptions()))
    {
        var defaultValueTest = context.DefaultValueTests.Single();
        Assert.Equal("test", defaultValueTest.Test);
    }
}
```

In [`TableInfo.LoadData`](https://github.com/borisdj/EFCore.BulkExtensions/blob/master/EFCore.BulkExtensions/TableInfo.cs#L229) this private property is not being filtered out in the Where expression. This leads to the exception when resolving the property with reflection in [Line 243](https://github.com/borisdj/EFCore.BulkExtensions/blob/master/EFCore.BulkExtensions/TableInfo.cs#L243). 

Apparently `GetDefaultValueSql()` returns `0` for EF Core 6 while it did return `null` on 3.1.
Looking at the condition of the Where expression I would guess that you found the same issue using Guids already but this issue probably will affect every non-nullable numeric type as well. I would have created a PR to fix this but since it could affect such a vast amount of types I'm unsure how to fix this issue properly.

**Comment by borisdj** [comment]
In line 243 try fix by adding just **'?'**
change from
`.GetProperty(propertyWithDefaultValue.Name)`
to
`.GetProperty(propertyWithDefaultValue.Name)?`

If it works you can make a PR.

**Comment by klemmchr** [comment]
My test runs through with that change, I cannot say whether that fully fixes it for all use cases. When making the PR, should I also remove that custom logic that is in place for Guids at the moment? I guess this is just there because of the same issue like I have.

**Comment by borisdj** [comment]
No, leave the rest, that was added for some other special use cases.

**Comment by klemmchr** [comment]
Thanks for the quick response. Any chance we could get an update for this soon?

**Comment by borisdj** [comment]
Published v6.3.4
Thx for Contrib.
